langchain_chat.py

```diff
@@ -0,0 +1,75 @@
+import streamlit as st
+from streamlit_chat import message
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.chat_models import ChatOpenAI
+from langchain.chains import ConversationalRetrievalChain
+from langchain.document_loaders.csv_loader import CSVLoader
+from langchain.vectorstores import FAISS
+import tempfile
+
+user_api_key = st.sidebar.text_input(
+    label="#### Your OpenAI API key 👇",
+    placeholder="Paste your openAI API key, sk-",
+    type="password")
+
+uploaded_file = st.sidebar.file_uploader("upload", type="csv")
+
+if uploaded_file :
+   #use tempfile because CSVLoader only accepts a file_path
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        tmp_file.write(uploaded_file.getvalue())
+        tmp_file_path = tmp_file.name
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The handling of the uploaded file can be improved. Instead of writing the uploaded file to a temporary file, consider modifying the CSVLoader to accept a file-like object. This would avoid unnecessary disk I/O operations and potential security risks associated with temporary files.
```diff
- if uploaded_file :
-    #use tempfile because CSVLoader only accepts a file_path
-     with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
-         tmp_file.write(uploaded_file.getvalue())
-         tmp_file_path = tmp_file.name
- 
-     loader = CSVLoader(file_path=tmp_file_path, encoding="utf-8", csv_args={
-                 'delimiter': ','})
+ if uploaded_file:
+     loader = CSVLoader(file=uploaded_file, encoding="utf-8", csv_args={'delimiter': ','})
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->